packages/components/src/form-token-field/token-input.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import type { ChangeEvent, ForwardedRef } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { WordPressComponentProps } from '../ui/context';
+import type { TokenInputProps } from './types';
+
+export function UnForwardedTokenInput(
+	props: WordPressComponentProps< TokenInputProps, 'input', false >,
+	ref: ForwardedRef< HTMLInputElement >
+) {
+	const {
+		value,
+		isExpanded,
+		instanceId,
+		selectedSuggestionIndex,
+		className,
+		onChange,
+		...restProps
+	} = props;
+
+	const size = value ? value.length + 1 : 0;
+
+	const onChangeHandler = ( event: ChangeEvent< HTMLInputElement > ) => {
+		if ( onChange ) {
+			onChange( {
+				value: event.target.value,
+			} );
+		}
+	};
+
+	return (
+		<input
+			ref={ ref }
+			id={ `components-form-token-input-${ instanceId }` }
+			type="text"
+			{ ...restProps }
+			value={ value || '' }
+			onChange={ onChange ? onChangeHandler : undefined }
```

Comment:
I guess we can remove this ternary now that we moved the check to the handler function itself?

```suggestion
			onChange={ onChangeHandler }
```